### PR TITLE
[gha] fix comment ops gh url

### PR DESCRIPTION
**Pull request by vulkoingim**
## Description

`gh` is not available in the self-hosted runner 🙄 
instead of installing it or including it in the runner's image, use the api to grab the run id.

also 🤮 
```
await new Promise(r => setTimeout(r, 5000));
```

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@vulkoingim: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: .github/workflows/comment-ops.yaml [hunk 1]
```diff
@@ -57,16 +57,27 @@ jobs:
...
-          sleep 5
-          url=$(gh run list --repo '${{ github.repository }}' --workflow build.yaml -b ${{ steps.comment-branch.outputs.head_ref }} --json event,url,status -q '[.[] | select(.event=="workflow_dispatch") | select(.status!="completed").url][0]')
-          {
-            echo "run_url=${url}"
-            echo "run_id=${url##*/}"
-          } >> $GITHUB_OUTPUT
+        with:
+          github-token: ${{ github.token }}
+          script: |
+            await new Promise(r => setTimeout(r, 5000));
```
Comment (mads-hartmann) [inline .github/workflows/comment-ops.yaml:65]: I'd love a comment on why this is needed ☺️

**Review by mads-hartmann — APPROVED** [review]
LGTM - I did have one note but can be done in a follow up PR if needed

File: .github/workflows/comment-ops.yaml (hunk 1 above)
Comment (vulkoingim) [reply .github/workflows/comment-ops.yaml:65]: It's in the comment above :)

```
      # Getting the ID of the workflow we triggered above is a bit tricky, as it's async :(
      # There is a trick we can do - add an input with a unique id, get the list of worklow runs, and check for that input in them until we can find it
      # But it's an overkill for now
      # Instead, we wait a little, and grab the last job that was triggered by a `workflow_dispatch` event, AND is not completed, and we hope for the best 🤞
```

**Pull request merged**